packages/teleterm/src/ui/services/clusters/clustersService.ts

```diff
@@ -383,6 +393,95 @@ export class ClustersService extends ImmutableStore<ClustersServiceState> {
     }
   }
 
+  async fetchServers(params: ServerSideParams) {
+    const cluster = this.state.clusters.get(params.clusterUri);
+    if (!cluster.connected) {
+      return;
+    }
+    return this.client.getServers(params);
+  }
+
+  async fetchDatabases(params: ServerSideParams) {
+    const cluster = this.state.clusters.get(params.clusterUri);
+    if (!cluster.connected) {
+      return;
+    }
+    return this.client.getDatabases(params);
+  }
+
+  getFetchCallback(clusterUri: string, resourceKind: ResourceKind) {
+    const cluster = this.state.clusters.get(clusterUri);
+    if (!cluster.connected) {
+      return;
+    }
+
+    switch (resourceKind) {
+      case 'node':
+        return this.client.getServers;
+      case 'db':
+        return this.client.getDatabases;
+      default: {
+        throw new Error(`Fetch not implemented for: ${resourceKind}`);
+      }
+    }
+  }
+
+  async getAccessRequests(clusterUri: string) {
+    const cluster = this.state.clusters.get(clusterUri);
+    if (!cluster.connected) {
+      return;
+    }
+
+    return this.client.getAccessRequests(clusterUri);
+  }
+
+  async deleteAccessRequest(clusterUri: string, requestId: string) {
+    const cluster = this.state.clusters.get(clusterUri);
+    if (!cluster.connected) {
+      return;
+    }
+    return this.client.deleteAccessRequest(clusterUri, requestId);
+  }
+
+  async assumeRole(
```

Comment:
Assuming resource access requests doesn't seem to work for me. After assuming such request, I don't see the requested ssh servers in Connect.

Are you able to reproduce that? Maybe it's a problem with my role setup. But I just double checked it against the docs and it seems alright.


https://user-images.githubusercontent.com/27113/192784482-29a4a335-822e-4ddc-a6e2-6f992af4ea98.mov

Reply:
Can you check if the issue persists on the web-ui as well? That is usually how I check if the issue is Connect related or my roles setup.

Reply:
I believe a refactor has messed up the _creation_ of the resource request actually after inspecting. Which breaks assuming. Will get a fix in shortly

Reply:
Yeah, I just checked it. I went to the Web UI and assumed the request that was created through Connect and I didn't see the resource in the Web UI.

I created another resource request through the Web UI, assumed it through the Web UI and I saw the resource.

Then I went back to Connect, assumed the request made through the Web UI and I could see the resource too.

Reply:
Just pushed a fix that should address this. I was accidentally passing in node name instead of cluster name during the create. You can see the difference by checking your incorrect resource request's "cluster name" in the review page and comparing to a newly created resource request after the fix. Thank you!

Reply:
Yup, it works now!